conda_forge_tick/git_utils.py

```diff
@@ -344,3 +346,46 @@ def is_github_api_limit_reached(e: github3.GitHubError, gh: github3.GitHub) -> b
         print(datetime.datetime.utcfromtimestamp(ts).strftime("%Y-%m-%dT%H:%M:%SZ"))
         return True
     return False
+
+
+def close_out_dirty_prs(
+    ctx: GithubContext,
+    pr_json: LazyJson,
+    gh: Optional[github3.GitHub] = None,
+    dry_run: bool = False,
+) -> Optional[dict]:
+    gh = ensure_gh(ctx, gh)
+    # run this twice so we always have the latest info (eg a thing was already closed)
+    if pr_json["state"] != "closed" and pr_json["mergeable_state"] == "dirty":
+        # update
+        if dry_run:
+            print("dry run: checking pr %s" % pr_json["id"])
+        else:
+            pr_obj = github3.pulls.PullRequest(dict(pr_json), gh)
+            pr_obj.refresh(True)
+            pr_json = pr_obj.as_dict()
+
+    if pr_json["state"] != "closed" and pr_json["mergeable_state"] == "dirty":
+        if dry_run:
+            print("dry run: comment and close pr %s" % pr_json["id"])
+        else:
+            if all(
+                c["commit"]["author"]["name"] in CF_BOT_NAMES for c in pr_obj.commits()
+            ):
+                pr_obj.create_comment(
+                    "I see that this PR has conflicts and I'm the only committer"
+                    "I'm going to close this PR and will make another one as"
+                    " appropriate. This was generated by {}".format(
+                        ctx.circle_build_url,
+                    ),
+                )
+                pr_obj.close()
+                delete_branch(ctx=ctx, pr_json=pr_json, dry_run=dry_run)
+                pr_obj.refresh(True)
+        d = pr_obj.as_dict()
+        # This will cause the update_nodes_with_bot_rerun to trigger properly and shouldn't be overridden since
+        # this is the last function to run, the long term solution here is to add the bot to conda-forge and then
+        # it should have label adding capability and we can just add the label properly
+        d["labels"].append("bot-rerun")
```

Comment:
how does this make it back to disk?

Reply:
Thank you!

Reply:
We return the dict, which is then used to update the pr information